Documentation: Add ServerContext, ServerConfiguration and GettingStartedWithDPFServer sections (#637)
pyansys/pydpf-core: docs/source/user_guide/getting_started_with_dpf_server.rst

simulation data. With DPF, you can perform complex preprocessing or postprocessing of large amounts of simulation data within a 
simulation workflow.

DPF Server is a package that contains all the necessary files to run the DPF Server, enabling DPF capabilities. It is available 
on the Ansys Customer Portal. DPF Server first available version is 6.0 (2023 R2).

For more information about DPF and its use, see :ref:`ref_user_guide`. 

The following section details how to use DPF Server package. For a quick start with DPF Server, see :ref:`ref_getting_started`. 

Installing DPF Server
---------------------

.. _target_installing_server:

#. Download the ansys_dpf_server_win_v<version_to_set>.zip or ansys_dpf_server_lin_v<version_to_set>.zip file as appropriate.
#. Unzip the package.
#. Change to the root folder (ansys_dpf_server_win_v<version_to_set>) of the unzipped package. 
#. In a Python environment, run the following command:

.. code::

    pip install -e . 
	
PyDPF-Core and PyDPF-Post python modules can now be used. The instructions to install and get started with PyDPF-Core 

> (ansys-dpf-core module) can be found at `Getting Started section <https://dpf.docs.pyansys.com/getting_started/install.html>`_. 

Author: Need to be updated looking at this PR: https://github.com/pyansys/pydpf-core/pull/642

Author: Should be ok

Using DPF Server
----------------

Running the DPF Server
~~~~~~~~~~~~~~~~~~~~~~

> On Windows, start the DPF Server by running the Ans.Dpf.Grpc.bat file in the unzipped package.

Reviewer: @anslpa should we say that these files are in dpf/bin/***x64/?

Author: I would like to avoid specifying this now (they might move)

simulation workflow.

DPF Server is a package that contains all the necessary files to run the DPF Server, enabling DPF capabilities. It is available 
on the Ansys Customer Portal. DPF Server first available version is 6.0 (2023 R2).

For more information about DPF and its use, see :ref:`ref_user_guide`. 

The following section details how to use DPF Server package. For a quick start with DPF Server, see :ref:`ref_getting_started`. 

Installing DPF Server
---------------------

.. _target_installing_server:

#. Download the ansys_dpf_server_win_v2023.2.pre0.zip or ansys_dpf_server_lin_v2023.2.pre0.zip file as appropriate.
#. Unzip the package.
#. Change to the root folder (ansys_dpf_server_win_v2023.2.pre0) of the unzipped package. 
#. In a Python environment, run the following command:

.. code::

    pip install -e . 
	
PyDPF-Core and PyDPF-Post python modules can now be used. The instructions to install and get started with PyDPF-Core 
(ansys-dpf-core module) can be found at `Getting Started section <https://dpf.docs.pyansys.com/getting_started/install.html>`_. 

Using DPF Server
----------------

Running the DPF Server
~~~~~~~~~~~~~~~~~~~~~~

> On Windows, start the DPF Server by running the Ans.Dpf.Grpc.bat file in the unzipped package.

Author: Done !

"ANSYS_DPF_ACCEPT_LA", you are expressing that you have a valid and existing license for the edition and version of DPF server you intend to use.

For a DPF Docker container usage, it can be set using:

.. code::

    docker run -e "ANSYS_DPF_ACCEPT_LA=Y" -e ANSYSLMD_LICENSE_FILE=1055@<license_server_to_use> -p 50052:50052 -e DOCKER_SERVER_PORT=50052 --expose=50052 dpf-core:v2023_2_pre0

For any other case, set "ANSYS_DPF_ACCEPT_LA" as an environment variable with "Y" value.

Replace "<license_server_to_use>" mention that ANSYSLMD_LICENSE_FILE environment variable points to the Ansys license server.
For more information about Ansys license mechanism use with DPF Server, see :ref:`Ansys licensing section<target_to_ansys_license_mechanism>`.

Ansys licensing
~~~~~~~~~~~~~~~

.. _target_to_ansys_license_mechanism:

DPF Server is protected by Ansys licensing mechanism.

DPF capabilities are available through the following main services: 
- Entry: Loads the minimum number of plugins for basic use. It is the default. It will only check if an Ansys License is available. 

> - Premium: Uses the premium DataProcessingCore.xml file to load most plugins.... with their environments. It will checkout a license. 

Reviewer 2: ```suggestion
- Premium: Uses the premium DataProcessingCore.xml file to load most plugins with their environments. It will checkout a license. 
```

> - Premium: Uses the premium DataProcessingCore.xml file to load most plugins.... with their environments. It will checkout a license. 

Author: @PProfizi I took that comment into account.

simulation workflow.

DPF Server is a package that contains all the necessary files to run the DPF Server, enabling DPF capabilities. It is available 
on the Ansys Customer Portal. DPF Server first available version is 6.0 (2023 R2).

For more information about DPF and its use, see :ref:`ref_user_guide`. 

The following section details how to use DPF Server package. For a quick start with DPF Server, see :ref:`ref_getting_started`. 

Installing DPF Server
---------------------

.. _target_installing_server:

#. Download the ansys_dpf_server_win_v2023.2.pre0.zip or ansys_dpf_server_lin_v2023.2.pre0.zip file as appropriate.
#. Unzip the package.
#. Change to the root folder (ansys_dpf_server_win_v2023.2.pre0) of the unzipped package. 
#. In a Python environment, run the following command:

.. code::

    pip install -e . 
	
PyDPF-Core and PyDPF-Post python modules can now be used. The instructions to install and get started with PyDPF-Core 
(ansys-dpf-core module) can be found at `Getting Started section <https://dpf.docs.pyansys.com/getting_started/install.html>`_. 

Using DPF Server
----------------

Running the DPF Server with PyDPF
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> With PyDPF-Core and PyDPF-Post, the creation of the first DPF entity will start the server.

Reviewer 2: ```suggestion
With PyDPF-Core and PyDPF-Post, the first creation of most DPF entities will start a DPF Server with the current default configuration and context.
For example, the following code will automatically start a DPF Server behind the scenes:

.. code::

    from ansys.dpf import core as dpf
    data_sources = dpf.DataSources()
```

> With PyDPF-Core, the server can be started using:

Reviewer 2: ```suggestion
With PyDPF-Core, you can also explicitly start a DPF Server using:
```

.. code::

    from ansys.dpf import core as dpf

>     dpf.start_local_server()

Reviewer 2: ```suggestion
    server = dpf.start_local_server()
```

> 

Reviewer 2: ```suggestion

To start a DPF Server from outside a Python environment, you can also use the execution script provided with your DPF Server package.
```
